src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,182 @@
+    {
+        showscale: scatterMarkerAttrs.showscale,
+        colorbar: colorbarAttrs,
+        shape: {
+            valType: 'enumerated',
+            values: ['straight', 'curved'],
+            dflt: 'curved',
+            role: 'info',
+            editType: 'plot',
+            description: 'Sets the shape of the paths'},
+    });
+
+module.exports = {
+    domain: {
+        x: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1, editType: 'calc'},
+                {valType: 'number', min: 0, max: 1, editType: 'calc'}
+            ],
+            dflt: [0, 1],
+            editType: 'calc',
+            description: [
+                'Sets the horizontal domain of this `parcats` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1, editType: 'calc'},
+                {valType: 'number', min: 0, max: 1, editType: 'calc'}
+            ],
+            dflt: [0, 1],
+            editType: 'calc',
+            description: [
+                'Sets the vertical domain of this `parcats` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        editType: 'calc'
+    },
+
+    tooltip: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Shows a tooltip when hover mode is `category` or `color`.'
+    },
+
+    hovermode: {
+        valType: 'enumerated',
+        values: ['none', 'category', 'color'],
+        dflt: 'category',
+        role: 'info',
+        editType: 'plot',
+        description: 'Sets the hover mode of the parcats diagram'
+    },
+    bundlecolors: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Sort paths so that like colors are bundled together'
+    },
+    sortpaths: {
+        valType: 'enumerated',
+        values: ['forward', 'backward'],
+        dflt: 'forward',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'If `forward` then sort paths based on dimensions from left to right.',
+            'If `backward` sort based on dimensions from right to left.'
+        ].join(' ')
+    },
+    // labelfont: fontAttrs({
+    //     editType: 'calc',
+    //     description: 'Sets the font for the `dimension` labels.'
+    // }),
+    //
+    // catfont: fontAttrs({
+    //     editType: 'calc',
+    //     description: 'Sets the font for the `category` labels.'
+    // }),
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        label: {
+            valType: 'string',
+            role: 'info',
+            editType: 'calc',
+            description: 'The shown name of the dimension.'
+        },
+        catDisplayInds: {
```

Comment:
(here and below: lowercase please)

I'm thinking `categoryorder` and `categoryarray` like we have for [cartesian axes](https://github.com/plotly/plotly.js/blob/0c1d6d5ebbd2334f712aede2a32236fb451838b7/src/plots/cartesian/layout_attributes.js#L734-L764) where you could initially specify one of the sort orders `'trace' | 'category ascending' | 'category descending'` for `categoryorder` but as soon as the user moves things around it would switch to `'array'` and create a `categoryarray`?

I guess `catValues` and `catLabels` allows you to use different data for `values` than you display... presumably the typical use here would be integer `values` and string `labels`? Makes sense, I'd just spell out the names completely: `categoryvalues` and `categorylabels`.

Reply:
Ah yes, this is a nice parallel with Cartesian axes.  To summarize:

 - `catValues` and `catDisplayInds` will be replaced by `categoryarray`
 - `categoryorder` will be introduced to control whether ordering is based on `categoryarray`, the `trace`, or the ascending/descending by category value. I believe the current default behavior is equivalent to `trace`, and having the  ascending and descending modes will be nice.
 - `categorylabels` will be introduced to control the category label displayed. This will be an array the same length as `categoryarray`, and will only have an effect when `categoryorder` is `array`.  Additionally, to keep things consistent, the `categorylabels` array will need to be reordered along with `categoryarray` when the user drags to reorder categories.

Reply:
Ok, I think this worked out really well.
Done in 7f90fc1, 5e60062, and 66c90fa (Sorry the commit history got a little messy here)